summingbird-storm/src/main/scala/com/twitter/summingbird/storm/builder/SpoutBuilder.scala

```diff
@@ -0,0 +1,105 @@
+package com.twitter.summingbird.storm.builder
+
+import com.twitter.algebird.Semigroup
+import com.twitter.summingbird.online.Externalizer
+import com.twitter.summingbird.option.JobId
+import EdgeType.AggregatedKeyValues
+import com.twitter.summingbird.{ Group, JobCounters, Name, SummingbirdRuntimeStats }
+import com.twitter.summingbird.storm._
+import com.twitter.tormenta.spout.Spout
+import org.apache.storm.metric.api.IMetric
+import org.apache.storm.task.TopologyContext
+import org.apache.storm.topology.IRichSpout
+import scala.collection.{ Map => CMap }
+
+private[builder] object SpoutBuilder {
+  def apply[T](
+    jobId: JobId,
+    spoutId: Topology.SpoutId[T],
+    spout: Topology.Spout[T],
+    outputEdges: List[Topology.Edge[T]]
+  ): IRichSpout = spout match {
+    case rawSpout: Topology.RawSpout[T] =>
+      createRawSpout(jobId, spoutId, rawSpout, outputEdges)
+    case keyValueSpout: Topology.KeyValueSpout[_, _] =>
+      createKeyValueSpout[Any, Any](
+        jobId,
+        spoutId.asInstanceOf[Topology.SpoutId[(Int, CMap[Any, Any])]],
+        keyValueSpout.asInstanceOf[Topology.KeyValueSpout[Any, Any]],
+        outputEdges.asInstanceOf[List[Topology.Edge[(Int, CMap[Any, Any])]]]
+      )(
+        keyValueSpout.semigroup.asInstanceOf[Semigroup[Any]]
+      )
+  }
+
+  def createRawSpout[T](
+    jobId: JobId,
+    spoutId: Topology.SpoutId[T],
+    spout: Topology.RawSpout[T],
+    outputEdges: List[Topology.Edge[T]]
+  ): IRichSpout = {
+    assert(
+      outputEdges.forall(_.edgeType.isInstanceOf[EdgeType.Item[_]]),
+      "Only `Item` edges supported from Tormenta Spout"
+    )
+
+    getStormSpout(jobId, spout.spout, spout.metrics)
+  }
+
+  def createKeyValueSpout[K, V: Semigroup](
+    jobId: JobId,
+    spoutId: Topology.SpoutId[(Int, CMap[K, V])],
+    spout: Topology.KeyValueSpout[K, V],
+    outputEdges: List[Topology.Edge[(Int, CMap[K, V])]]
+  ): IRichSpout = {
+    val stormSpout = getStormSpout(jobId, spout.spout, spout.metrics)
+    assert(outputEdges.nonEmpty, "Should be at least one output edge from KeyValueSpout")
+    assert(
+      outputEdges.forall(edge => edge.edgeType.isInstanceOf[AggregatedKeyValues[K, V]]),
+      "All output edges from KeyValueSpout should be `AggregatedKeyValues`"
+    )
+    val shards = outputEdges.head.edgeType.asInstanceOf[AggregatedKeyValues[K, V]].shards
```

Comment:
Assumption is there are outputEdges, which is true now but not be in future if we support Spout only topologies. May be add an assert for the assumption.